### PR TITLE
Adds enchantgrade chance per refine

### PR DESCRIPTION
**Pull request by datawulf**
* **Addressed Issue(s)**: 

https://github.com/rathena/rathena/issues/7607

* **Server Mode**:  Renewal

* **Description of Pull Request**: 

adds enchantgrade chance per refine of item

### CODE REVIEW

File: src/map/status.cpp [hunk 1]
```diff
@@ -676,14 +676,18 @@ uint64 EnchantgradeDatabase::parseBodyNode( const ryml::NodeRef& node ){
...
+			if( this->nodeExists( gradeNode, "Chances" ) ){
+				uint16 chance,refineIndex;
 
-				if( !this->asUInt16Rate( gradeNode, "Chance", chance ) ){
-					return 0;
+				for( const ryml::NodeRef& chanceNode : gradeNode["Chances"] ){
+					if( !this->asUInt16( chanceNode, "Refine", refineIndex ) ){
+						return 0;
+					}
+					if( !this->asUInt16( chanceNode, "Chance", chance ) ){
```
Comment (aleos89) [inline src/map/status.cpp:686]:
```suggestion
					if( !this->asUInt16Rate( chanceNode, "Chance", chance ) ){
```
This keeps it between 0~10000.

File: src/map/status.cpp [hunk 2]
```diff
@@ -676,14 +676,18 @@ uint64 EnchantgradeDatabase::parseBodyNode( const ryml::NodeRef& node ){
...
 
-			if( this->nodeExists( gradeNode, "Chance" ) ){
-				uint16 chance;
+			if( this->nodeExists( gradeNode, "Chances" ) ){
+				uint16 chance,refineIndex;
 
-				if( !this->asUInt16Rate( gradeNode, "Chance", chance ) ){
-					return 0;
+				for( const ryml::NodeRef& chanceNode : gradeNode["Chances"] ){
+					if( !this->asUInt16( chanceNode, "Refine", refineIndex ) ){
```
Comment (aleos89) [inline src/map/status.cpp:683]: Probably will also want to add a check here to make sure this doesn't exceed MAX_REFINE.

**Comment by eppc0330** [comment]
Great work!

File: db/re/enchantgrade.yml [hunk 3]
```diff
@@ -28,7 +28,9 @@
 #       Grades:                         Enchantgrade settings per grade level.
 #         - Grade                       Enchantgrade level.
 #           Refine                      Required refine level.
```
Comment (Atemo) [inline db/re/enchantgrade.yml:30]: This field become useless and could be removed, I guess?

**Review by Atemo — APPROVED** [review]
It looks good !

**Review by aleos89 — APPROVED** [review]

**Pull request merged**